test_discovery.py

```diff
@@ -62,10 +62,27 @@
 """
 
 
+def extract_schema_results_as_entities(
+    test_dict, intent_results, entity_results
+):
+    for k, v in json.loads(test_dict['schema']).items():
+        intent_results["expected_entities"]["schema_" + k] = v
+        intent_results["observed_entities"]["schema_" + k] = v
+
+    schema_failures = [
+        f[4] for f in entity_results['test_failures'] if f[2] == "schema"
+    ]
+
+    for k, v in json.loads(test_dict['schema']).items():
```

Comment:
same as above

Reply:
In this case, we specifically need the key value pairs.

Reply:
I see what you mean now. This is a typo — it should be iterating over the schema failures